homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -0,0 +1,101 @@
+"""Interfaces with TotalConnect sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_DOOR,
+    DEVICE_CLASS_GAS,
+    DEVICE_CLASS_SMOKE,
+    BinarySensorDevice,
+)
+
+from . import DOMAIN as TOTALCONNECT_DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up a sensor for a TotalConnect device."""
+    if discovery_info is None:
+        return
+
+    sensors = []
+
+    client_locations = hass.data[TOTALCONNECT_DOMAIN].client.locations
+
+    for location_id in client_locations:
+        for zone in client_locations[location_id].zones:
+            sensors.append(
+                TotalConnectBinarySensor(zone, location_id, client_locations)
+            )
+    add_entities(sensors)
+
+
+class TotalConnectBinarySensor(BinarySensorDevice):
+    """Represent an TotalConnect zone."""
+
+    def __init__(self, zone_id, location_id, locations):
+        """Initialize the TotalConnect status."""
+        self._zone_id = zone_id
+        self._location_id = location_id
+        self._zone = locations[location_id].zones[zone_id]
```

Comment:
Please get the zone before you create an instance of this class and pass it into the constructor as a parameter.

Reply:
concur